converter.go

```diff
@@ -818,24 +959,137 @@ func mapStructuredArrayNativeArrowRows[T any](offsets []int32, rowIdx int, creat
 	return arr, nil
 }
 
+func extractInt64(values arrow.Array, j int) (int64, error) {
+	switch typedValues := values.(type) {
+	case *array.Decimal128:
+		return int64(typedValues.Value(j).LowBits()), nil
+	case *array.Int64:
+		return typedValues.Value(j), nil
+	case *array.Int32:
+		return int64(typedValues.Value(j)), nil
+	case *array.Int16:
+		return int64(typedValues.Value(j)), nil
+	case *array.Int8:
+		return int64(typedValues.Value(j)), nil
+	}
+	return 0, fmt.Errorf("unsupported map type: %T", values.DataType().Name())
+}
+
+func buildStructuredMapFromArrow[K comparable](rowIdx int, srcColumnMeta execResponseRowType, offsets []int32, keyFunc func(j int) (K, error), items arrow.Array, higherPrecision bool, loc *time.Location) (snowflakeValue, error) {
+	switch srcColumnMeta.Fields[1].Type {
+	case "text":
+		return mapStructuredMapNativeArrowRows(make(map[K]string), offsets, rowIdx, keyFunc, func(j int) (string, error) {
+			return items.(*array.String).Value(j), nil
+		})
+	case "boolean":
+		return mapStructuredMapNativeArrowRows(make(map[K]bool), offsets, rowIdx, keyFunc, func(j int) (bool, error) {
+			return items.(*array.Boolean).Value(j), nil
+		})
+	case "fixed":
```

Comment:
I see that in case of "fixed" actual logic differs only by type correct? can we remove the code duplication? should be fairly easy to extract common logic

Reply:
Done.